src/hooks/encounter/user.ts

```diff
@@ -3,41 +3,44 @@ import { useCallback, useContext, useEffect, useState } from 'react';
 import { showUserAsync } from '../../encounterApi';
 import { getEncounterUserId } from '../../helpers';
 import { NetworkContext } from '../../NetworkProvider';
-import { UserWithId } from '../../types';
+import { User } from '../../types';
 
-const loadUser = async () => {
-  const userId = await getEncounterUserId();
+const loadUser = async (userId: string) => {
+  const user = await showUserAsync(userId);
 
-  if (userId) {
-    const user = await showUserAsync(userId);
-
-    if (user) {
-      return user;
-    }
+  if (user) {
+    return user;
   }
 
-  throw new Error('Error while loading user: ' + userId ?? 'Missing user id');
+  throw new Error('Error while loading user: ' + userId);
 };
 
 export const useEncounterUser = (): {
   error: boolean;
   loading: boolean;
   refresh: () => void;
   refreshing: boolean;
-  user?: UserWithId;
+  user?: User;
+  userId?: string;
 } => {
   const { isConnected, isMainserverUp } = useContext(NetworkContext);
   const [error, setError] = useState(false);
   const [loading, setLoading] = useState(true);
-  const [user, setUser] = useState<UserWithId>();
+  const [user, setUser] = useState<User>();
+  const [userId, setUserId] = useState<string>();
   const [refreshing, setRefreshing] = useState(false);
 
   const loadUserCallback = useCallback(async () => {
     setRefreshing(true);
     setError(false);
 
     try {
-      setUser(await loadUser());
+      const storedId = await getEncounterUserId();
```

Comment:
`storedId` regarded out of context very generic. as we handle different ids (user, encounters, ...) this can lead to misunderstanding if not traversing the code, maybe... `const storedEncounterUserId = await getEncounterUserId();` or `const storedUserId = await getEncounterUserId();` would be more precise, right?